PROTO_tests/tests/client_dtypes_test.py

```diff
@@ -0,0 +1,181 @@
+        names = ["8", "4", "2", "1"]
+        f = ak.Fields(values, names)
+        assert isinstance(f, ak.Fields)
+        assert f.to_list() == ["---- (0)", "---1 (1)", "--2- (2)", "--21 (3)"]
+        assert f.dtype == ak.bitType
+
+        # Named fields with reversed bit order
+        values = ak.array([0, 1, 5, 8, 12])
+        names = ["Bit1", "Bit2", "Bit3", "Bit4"]
+        f = ak.Fields(values, names, MSB_left=False, separator="//")
+        expected = [
+            "----//----//----//----// (0)",
+            "Bit1//----//----//----// (1)",
+            "Bit1//----//Bit3//----// (5)",
+            "----//----//----//Bit4// (8)",
+            "----//----//Bit3//Bit4// (12)",
+        ]
+        assert f.to_list() == expected
+
+        values = ak.arange(8, dtype=ak.uint64)
+        names = [f"Bit{x}" for x in range(65)]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", "t"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", ""]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["abc", "123"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, separator="abc")
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, pad="|!~", separator="//")
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("int_types", INT_TYPES)
+    def test_ipv4_creation(self, prob_size, int_types):
+        ips = ak.randint(1, 2**32, prob_size)
+        ip_list = ak.array(ips, dtype=int_types)
+        ipv4 = ak.IPv4(ip_list)
+        py_ips = [ipaddress.IPv4Address(ip).compressed for ip in ips.to_list()]
+
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == py_ips
+        assert ipv4.dtype == ak.bitType
+
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(f"{ips[0]}")
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(ak.array([ips[0] + 0.177]))
+
+        # Test handling of python dotted-quad input
+        ipv4 = ak.ip_address(py_ips)
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == py_ips
+        assert ipv4.dtype == ak.bitType
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_ipv4_normalization(self, prob_size):
+        prob_size = 100
+        ips = ak.randint(1, 2**32, prob_size)
+        ip_list = ak.array(ips)
+        ipv4 = ak.IPv4(ip_list)
+        ip_as_dot = [ipaddress.IPv4Address(ip) for ip in ips.to_list()]
+        ip_as_int = [ipv4.normalize(ipd) for ipd in ip_as_dot]
+        assert ips.to_list() == ip_as_int
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_is_ipv4(self, size):
+        x = [random.getrandbits(32) for i in range(size)]
```

Comment:
half these tests parameterize `pytest.prob_size` as `prob_size` and the other half as `size`, we should pick one and try and be consistent.

Also I feel like `x` could get quite large for the client to handle depending on the value of `size`. I think it'd be safer to just pick a fixed size. Especially because we have several `.to_list` calls later in the test

Also `i` is never used, so we should replace it with `_` like what was suggested previous for `test_is_ipv6`

Reply:
Updated